omegaconf/listconfig.py

```diff
@@ -101,18 +102,26 @@ def _validate_set(self, key: Any, value: Any) -> None:
                     )
 
         target_type = self._metadata.element_type
+        if target_type in (Any, None):
+            return
+
         value_type = OmegaConf.get_type(value)
-        if is_structured_config(target_type):
-            if (
-                target_type is not None
-                and value_type is not None
-                and not issubclass(value_type, target_type)
-            ):
-                msg = (
-                    f"Invalid type assigned : {type_str(value_type)} is not a "
-                    f"subclass of {type_str(target_type)}. value: {value}"
-                )
-                raise ValidationError(msg)
+
+        if (
+            (
+                is_structured_config(value_type)
+                or is_structured_config(target_type)
+                or isinstance(value, ValueNode)
+            )
+            and value_type is not None
+            and not issubclass(value_type, target_type)
+        ):
+            msg = (
+                f"Invalid type assigned : {type_str(value_type)} is not a "
+                f"subclass of {type_str(target_type)}. value: {value}"
+            )
+
+            raise ValidationError(msg)
 
```

Comment:
This still looks too complicated.
can you try to make it more readable by refacotring it a bit? (maybe extract local boolean values and use them in conditions like in DictConfig).